src/Raven.Server/Utils/PoolOfThreads.cs

```diff
@@ -270,21 +270,21 @@ private bool ResetThreadAffinity()
 
             private static bool ResetThreadPriority()
             {
-                try
-                {
-                    if (Thread.CurrentThread.Priority != ThreadPriority.Normal)
-                        Thread.CurrentThread.Priority = ThreadPriority.Normal;
-                    return true;
-                }
-                catch (Exception e)
+                if (Thread.CurrentThread.Priority != ThreadPriority.Normal)
                 {
-                    // if we can't reset it, better just kill it
-                    if (_log.IsInfoEnabled)
+                    if (ThreadHelper.TrySetThreadPriority(ThreadPriority.Normal, null, _log) == false)
                     {
-                        _log.Info($"Unable to set this thread priority to normal, since we don't want its priority of {Thread.CurrentThread.Priority}, we'll let it exit", e);
+                        // if we can't reset it, better just kill it
+                        if (_log.IsInfoEnabled)
+                        {
+                            _log.Info($"Unable to set this thread priority to normal, since we don't want its priority of {Thread.CurrentThread.Priority}, we'll let it exit");
+                        }
+
+                        return false;
```

Comment:
This is wrong. Even if we can't set the priority, we want to keep it around

Reply:
This was taken from the original code. https://github.com/ravendb/ravendb/blob/f8f640b39728bd8b41c82a3e5241e032cb52eac0/src/Raven.Server/Utils/PoolOfThreads.cs#L284
If we cannot set the priority to `Normal`, we better use another thread, no? (in order not to use a thread that we already set its Priority)

Reply:
Hm... I guess so, then.